src/DynamoCore/Utilities/NotificationCounter.cs

```diff
@@ -0,0 +1,235 @@
+﻿using HarmonyLib;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Timers;
+
+namespace Dynamo.Utilities
+{
+    internal class NotificationLog
+    {
+        internal SortedSet<int> objects = new SortedSet<int>();
+        internal int counter;
+    }
+
+    internal static class NotificationCounter
```

Comment:
I think this name is still confusing for devs working on notifications and will be in the future as well, especially since this class has no summary.